store/src/java/com/zimbra/cs/account/ldap/entry/LdapDataSource.java

```diff
@@ -62,10 +62,8 @@ public static String getObjectClass(DataSourceType type) {
                 return AttributeClass.OC_zimbraRssDataSource;
             case gal:
                 return AttributeClass.OC_zimbraGalDataSource;
-            case custom:
-                return AttributeClass.OC_zimbraDataSource;
             default: 
-                return null;
+                return AttributeClass.OC_zimbraDataSource;
```

Comment:
Nothing for `oauth`?

Reply:
correct, oauth and everything else should return "dataSource" objectClass. What I am trying to accomplish here is to avoid using special LDAP objectClass for every kind of data source and use a generic dataSource LDAP class while moving al the specifics into implementation. I think that all the subclasses of LDAP objectClass "dataSource" are a design flaw and shouldn't have been added.

Reply:
Could you add a comment on your reasoning there please?  To avoid someone going in the opposite direction in the future :)